Add documentation about the new values in trigger auth in PostgreSQL Scaler (#535)
kedacore/keda-docs: content/docs/2.5/scalers/postgresql.md

> arguments (host, userName, password, etc.), and the scaler will form a connection string

Reviewer: Nit - Can you ensure all values use the `PostgreSQL`  casing please given it's a product name?

Reviewer: Same for the `<name>:` where the `:` needs to be moved out.

Reviewer: Actually let's align with the majority of scalers:
> - `host` - The hostname of the MSSQL instance endpoint.

Author: Do you mean using `propery - description`? Prometheus Scaler for example uses `property: description` also. WDYT if in another PR I review the scalers docs and unify the notation?

Reviewer: Yes, that notation.

A seperate PR would be awesome, if you are willing to do that of course.

Author: Sure, I will do :)

Author: I have opened it
https://github.com/kedacore/keda-docs/discussions/539

internally.

> - `host:` Service URL to postgresql. Note that you should use a fully qualified URL -- including the namespace -- as KEDA will need to contact postgresql from a different namespace.

Reviewer: ```suggestion
- `host:` Service URL to PostgreSQL. Note that you should use a fully qualified URL (including the namespace) as KEDA will need to contact PostgreSQL from a different namespace.
```

**Password Authentication:**

> - `host:` Service URL to postgresql. Note that you should use a full svc URL as KEDA will need to contact postgresql from a different namespace

Reviewer 2: ```suggestion
- `host:` Service URL to postgresql. Note that you should use a fully qualified URL -- including the namespace -- as KEDA will need to contact postgresql from a different namespace
```

Author: Honestly, I have copied and pasted from the current "metadata definitions", so I will change both :)

> - `host:` Service URL to postgresql. Note that you should use a fully qualified URL -- including the namespace -- as KEDA will need to contact postgresql from a different namespace.

Reviewer: See above

> - `host:` Service URL to postgresql. Note that you should use a fully qualified URL -- including the namespace -- as KEDA will need to contact postgresql from a different namespace.

Reviewer: ```suggestion
- `host`: Service URL to postgresql. Note that you should use a fully qualified URL -- including the namespace -- as KEDA will need to contact postgresql from a different namespace.
```

> - `userName:` Username for postgresql user

Reviewer: ```suggestion
- `userName`: Username for postgresql user
```

> - `password` Password for configured user to login to postgreSQL database variables.

Reviewer: ```suggestion
- `password`: Password for configured user to login to postgreSQL database.
```

> - `port:` Postgresql port

Reviewer: ```suggestion
- `port`: Port of PostgreSQL database
```

> - `dbName:` Postgresql Database name

Reviewer: ```suggestion
- `dbName`: Name of PostgreSQL database
```

> - `sslmode:` SSL policy for communicating with database

Reviewer: ```suggestion
- `sslmode`: SSL policy for communicating with database
```

Is the field `sslmode` or `sslMode`?

Author: In the connection string is `sslMode` but our trigger variable is [`sslmode`](https://github.com/kedacore/keda/blob/main/pkg/scalers/postgresql_scaler.go#L105)

Reviewer: Can you elaborate on that please? You mean trigger metadata uses `sslMode` but trigger authentication uses `sslmode`?

Reviewer: No, the other way around probably?

Author: The required key in the connection string is `sslMode`, but inside the scaler we are looking for `sslmode` when we are parsing the metadata. Our key is `sslmode`, we take the value from the metadata key `sslmode` and place in the connection string key `sslMode`

Author: Maybe we can look for both when we are parsing the scaler metadata and deprecate `sslmode` in favor of `sslMode` (the last one is how we have to place in connection string)

Reviewer: If `sslMode` is only in the connection string then we can just keep on using `sslmode` as we do today, no need to check for `sslMode` field then as well

Author: okey, so basically we can let it as it is :)
